halo2-base/src/gates/range.rs

```diff
@@ -340,15 +411,40 @@ pub trait RangeInstructions<F: ScalarField> {
     }
 }
 
+/// Represents a chip that can be used to constrain a field element number `x` is within a certain number of bits.
 #[derive(Clone, Debug)]
 pub struct RangeChip<F: ScalarField> {
+    /// # RangeChip
+    /// TODO: Add more to this and make it less clunky
+    /// TODO: Add example
+    /// Provides methods to constrain a field element number `x` is within a certain number of bits.
+    /// Declares a LookUp table of [0, 2**lookup_bits) and constrains whether a field element appears in this table
+    /// 
+    /// # LookUp bits:
+    /// The environment variable DEGREE specifies that the circuit will have 2**DEGREE rows. 
```

Comment:
I don't think `DEGREE` is actually needed here, it was more of a helper. See the definition of `GateConfig` for what the config parameters would actually be.
Similar for `LOOKUP_BITS`: we only need to pass in `lookup_bits`